File: benchmarks/python/array_push_sum.py
def main() -> None:
    values: list[int] = []
    seed = 1
    for i in range(30000):
        values.append((i * 31 + seed) % 997)

    total = 0
    for i, value in enumerate(values):
        total += value * (i % 13)

    print(total)


if __name__ == "__main__":
    main()

File: benchmarks/c/array_push_sum.c
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    int64_t *values = malloc(30000 * sizeof(*values));
    int64_t seed = 1;
    int64_t total = 0;

    if (values == NULL) {
        return 1;
    }

    for (int64_t i = 0; i < 30000; ++i) {
        values[i] = (i * 31 + seed) % 997;
    }

    for (int64_t i = 0; i < 30000; ++i) {
        total += values[i] * (i % 13);
    }

    free(values);
    printf("%lld\n", (long long) total);
    return 0;
}
